src/classes/CCON_Faculty_TDTM.cls

```diff
@@ -96,29 +95,43 @@ public with sharing class CCON_Faculty_TDTM extends TDTM_Runnable {
                     for (Integer i = 0; i < newlist.size(); i++) {
                         Course_Enrollment__c enroll = (Course_Enrollment__c)newlist[i];
 
-                        if(enroll.get('RecordTypeId') == facultyRecType) {
-                            enroll.Account__c = offerings.get(enroll.Course_Offering__c).Course__r.Account__c;
-
-                            if (contactIdAccountIdAfflList.containsKey(enroll.Contact__c) && (
-                                 contactIdAccountIdAfflList.get(enroll.Contact__c).containsKey(enroll.Account__c))) {
-                                enroll.Affiliation__c = contactIdAccountIdAfflList.get(enroll.Contact__c).get(enroll.Account__c)[0].Id;
-                            } else {
-                                Affiliation__c affl = new Affiliation__c(
-                                    Contact__c = enroll.Contact__c,
-                                    Account__c = enroll.Account__c,
-                                    Role__c = 'Faculty'
-                                );
-                                affiliationsToInsert.add(affl);
-                                connsNeedingAffls.add(enroll);
+                        // Skip over the Course Connection if it is not for a Faculty.
+                        if(enroll.get('RecordTypeId') != facultyRecType) {
+                            continue;
+                        }
+
+                        enroll.Account__c = offerings.get(enroll.Course_Offering__c).Course__r.Account__c;
+
+                        if (contactIdAccountIdAfflList.containsKey(enroll.Contact__c) && (
+                                contactIdAccountIdAfflList.get(enroll.Contact__c).containsKey(enroll.Account__c))) {
+                            enroll.Affiliation__c = contactIdAccountIdAfflList.get(enroll.Contact__c).get(enroll.Account__c)[0].Id;
+                        } else {
+                            connsNeedingAffls.add(enroll);
+                            
+                            // Is the Affiliation already being inserted? If so, do not insert a duplicate.
+                            String contactAccountKey = enroll.Contact__c + '' + enroll.Account__c;
+                            if (affiliationsToInsertByContactAccountString.containsKey(contactAccountKey)) {
```

Comment:
a big part of me wishes this didn't work so damn well because I hate reading it, but I'll be damned if its not much more efficient :). Actually we should consider doing this more, its something I rarely think to do so kudos to you.

Reply:
Yeah I was taught to exit early when possible (or continue in this case)... it usually results in less nested if statements.

````
if (true condition) {
    // logic here, with its own potential nesting
}

// vs

if (false condition) {
    exit
}
// logic here instead
````

Reply:
Yea, I'm always thrown by the continue/break stuff. For me perhaps in reminds of a GOTO statement and feels like I'm forced to jump about in the code. That said, I should probably get used to it because I feel its efficient. I just hate it :)